src/provider/provider_tracking.c

```diff
@@ -76,6 +127,96 @@ static umf_result_t umfMemoryTrackerAdd(umf_memory_tracker_handle_t hTracker,
     return UMF_RESULT_ERROR_UNKNOWN;
 }
 
+static tracker_alloc_info_t *
+critnib_get_last(umf_memory_tracker_handle_t hTracker, const void *ptr,
+                 int *_level, tracker_alloc_info_t **_prev_value) {
+    assert(ptr);
+
+    tracker_alloc_info_t *prev_value = NULL;
+    tracker_alloc_info_t *rvalue = NULL;
+    uintptr_t rkey = 0;
+    int level = 0;
+    int found = 0;
+
+    do {
+        found =
+            critnib_find(hTracker->alloc_segments_map[level], (uintptr_t)ptr,
+                         FIND_LE, (void *)&rkey, (void **)&rvalue);
+        if (found && (uintptr_t)ptr < rkey + rvalue->size) {
+            if (rvalue->n_children) {
+                if (level == MAX_LEVELS_OF_MAPS - 1) {
+                    break;
+                }
+                level++;
+                prev_value = rvalue;
+            }
+        }
+    } while (found && ((uintptr_t)ptr < rkey + rvalue->size) &&
+             rvalue->n_children);
+
+    if (!rvalue || rkey != (uintptr_t)ptr) {
+        return NULL;
+    }
+
+    if (_level) {
+        *_level = level;
+    }
+    if (_prev_value) {
+        *_prev_value = prev_value;
+    }
+
+    return rvalue;
+}
+
+static tracker_alloc_info_t *
+critnib_get_last_no_children(umf_memory_tracker_handle_t hTracker,
```

Comment:
the same is here, pleas do not use `critnib` in the name.

Reply:
Done